actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -161,28 +162,17 @@ def assert_enqueued_email_with(mailer, method, params: nil, args: nil, queue: Ac
         mailer = mailer.instance_variable_get(:@mailer)
       end
 
-      expected_args = if args.is_a?(Proc) || params.is_a?(Proc)
-        -> (job_args) {
-          # Ordered from the least to the most heavy checks.
-          return false if [mailer.to_s, method.to_s, "deliver_now"] != job_args[0..2]
-          return false if !params.is_a?(Proc) && params != job_args[3][:params]
-          return false if !args.is_a?(Proc) && Array(args) != job_args[3][:args]
-          return false if args.is_a?(Proc) && !args.(job_args[3][:args])
-          return false if params.is_a?(Proc) && !params.(job_args[3][:params])
+      params, args = args, nil if args.is_a?(Hash)
+      args = Array(args) unless args.is_a?(Proc)
 
-          true
-        }
-      else
-        if args.is_a?(Hash)
-          [mailer.to_s, method.to_s, "deliver_now", params: args, args: []]
-        elsif params.present?
-          [mailer.to_s, method.to_s, "deliver_now", params: params, args: Array(args)]
-        else
-          [mailer.to_s, method.to_s, "deliver_now", args: Array(args)]
-        end
+      expected = ->(job_args) do
+        job_kwargs = job_args.extract_options!
+
+        [mailer.to_s, method.to_s, "deliver_now"] == job_args &&
+          params === job_kwargs[:params] && args === job_kwargs[:args]
```

Comment:
Great use of `===`. 👍🏻 If left arg is an array/hash it just compares them, if proc, it passes the rhs as the proc arg, and gets the truthy/falsy return.